pulsar-client-cpp/lib/ConsumerImpl.cc

```diff
@@ -1229,7 +1244,17 @@ void ConsumerImpl::seekAsync(const MessageId& msgId, ResultCallback callback) {
         LOG_DEBUG(getName() << " Sending seek Command for Consumer - " << getConsumerId() << ", requestId - "
                             << requestId);
         Future<Result, ResponseData> future =
-            cnx->sendRequestWithId(Commands::newSeek(consumerId_, requestId, msgId), requestId);
+            msgId.isChunkMessageid()
+                ? cnx->sendRequestWithId(
+                      Commands::newSeek(consumerId_, requestId,
+                                        std::dynamic_pointer_cast<ChunkMessageIdImpl>(msgId.impl_)
+                                            ->getFirstChunkMessageIdImpl()
```

Comment:
Do you make sure the casted pointer is not null?

Reply:
When ChunkMessageIdImpl be constructed, it must need the param of FirstChunkMessageIdImpl, so i think dont need to check here

Reply:
It's not explicit. Your assumption is based on the fact that `msgId.isChunkMessageid()` is equivalent to `std::dynamic_pointer_cast<ChunkMessageIdImpl>(msgId.impl_) != null`. Once the implementation of `MessageId::isChunkMessageid()` changed, the assumption would not be right.

It's better not to call `isChunkMessageid()` internally. Just use

```c++
auto chunkMessageId = std::dynamic_pointer_cast<ChunkMessageIdImpl>(msgId.impl_);
if (chunkMessageId) {
    /* do sth with `chunkMessageId`... */
}
```

instead of

```c++
if (msgId.isChunkMessageid()) {
    auto chunkMessageId = std::dynamic_pointer_cast<ChunkMessageIdImpl>(msgId.impl_);
    /* do sth with `chunkMessageId`... */
}
```

Reply:
BTW, do we need to expose the `isChunkMessageid()` to users? Even if the user knew the message ID is a chunked message ID by calling `isChunkMessageid()`, there is no way to get the underlying `ChunkMessageIdImpl`.

Reply:
Could you answer my questions here?

Reply:
sorry, i have changed it already

Reply:
OK, I have deleted all exposed `isChunkMessageid()` interfaces to prevent problems if interface changes